helper/validation/map_test.go

```diff
@@ -0,0 +1,209 @@
+package validation
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+)
+
+func TestValidationMapKeyLenBetween(t *testing.T) {
+	cases := map[string]struct {
+		Value         interface{}
+		ExpectedDiags diag.Diagnostics
+	}{
+		"TooLong": {
+			Value: map[string]interface{}{
+				"ABC":    "123",
+				"UVWXYZ": "123456",
+			},
+			ExpectedDiags: diag.Diagnostics{
+				{
+					Severity:      diag.Error,
+					AttributePath: append(cty.Path{}, cty.IndexStep{Key: cty.StringVal("UVWXYZ")}),
+				},
+			},
+		},
+		"TooShort": {
+			Value: map[string]interface{}{
+				"ABC": "123",
+				"U":   "1",
+			},
+			ExpectedDiags: diag.Diagnostics{
+				{
+					Severity:      diag.Error,
+					AttributePath: append(cty.Path{}, cty.IndexStep{Key: cty.StringVal("U")}),
+				},
+			},
+		},
+		"AllGood": {
+			Value: map[string]interface{}{
+				"AB":    "12",
+				"UVWXY": "12345",
+			},
+			ExpectedDiags: nil,
+		},
+	}
+
+	fn := MapKeyLenBetween(2, 5)
+
+	for tn, tc := range cases {
+		t.Run(tn, func(t *testing.T) {
+			diags := fn(tc.Value, cty.Path{})
+
+			if len(diags) != len(tc.ExpectedDiags) {
+				t.Fatalf("%s: wrong number of diags, expected %d, got %d", tn, len(tc.ExpectedDiags), len(diags))
+			}
+			for j := range diags {
+				if diags[j].Severity != tc.ExpectedDiags[j].Severity {
```

Comment:
Seeing test cases around diags makes me think it should have a `.Equals()` helper. to reduce boilerplate like this for devs. You could PR that helper in with this PR if you like. Or I can add it and you rebase.